private_set_intersection/javascript/src/__tests__/client.test.ts

```diff
@@ -45,39 +44,137 @@ describe('PSI Client', () => {
-
-    const intersection = client.processResponse(serverSetup, serverResponse)
-
+    const serverSetup = JSON.stringify({
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      num_hash_functions: 14,
+      bits:
+        '1PnolAVduzqn+tXnoeyZ0ztdnwvAidt0U7gc2rWCtP5i6BDtSUydixXwGHP2mVAZ8PCNqGuTWUQO5VB0OEHTH8Cr1JhLxBDHRM4fAPEUvjjkBPNpsPRRtqzN2fW8y7rcANEwaRyInPsw1t4KYB0Q5vdy38VlPB1/Bz/adfGoGnXQ4fPNI/PN5s+zLcPev9odjAWyUQms8bvPHHsL2uyS7Uu5LRtOfdfYX5zWilxzNXECrabnnUnMqD5NH8c7oreTCWxwRQfYMxhf5K/EZTiTHphCi9O5Ey4ahwvNvykbvXdHQ7CDnLTKmomN6su5BrUt'
+    })
+    const serverResponse = JSON.stringify({
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      encrypted_elements: [
+        'Aw8M6ufg9yUnMdHXQKqZ+2g4Z2uhLzn2J/7dKnp/uDEL',
+        'Av/dJyi/E7gunWLvXX2A1AcnU6dEmB40igRCrqskGTgB',
+        'Av8i8ltywvz4VRFTFju//VoavSL5Fmt95wWd8ChPCrCS',
+        'A1kEdR6hNbGz1sQfHSInQrbZfpDTu6OPiQbnEisAA1wl',
+        'Ap/z+I2cPqyUxM0YYlLOpKtRJYaSMfgNcyRY1Jynun0r',
+        'A6CpgzV4+F6Jjr+yOYbLdpkphMVTjoRuiZOP8uI+7he3',
+        'AnQOkG1+kkYuMvtOcQPqV5YKTYoVe/P+/cUHrfaWDJaD',
+        'An8POEQwL/cEHmQrwAdoBSPYh6a6jPOthH3vBx+2owOF',
+        'A23B3m3hJtAi607Z8FKF/P4fXCTdMCPzOjxgCFlna9K6',
+        'AqV+jTb23uTHPDQmPZsEuxDm5kKYIuI9pkEjvM8fMTG9'
+      ]
+    })
+    const intersection = client.getIntersectionSize(serverSetup, serverResponse)
     expect(intersection).toStrictEqual(0)
   })
 
-  test('It should throw if attempting to process a response after deletion', async () => {
+  test('It should process a response (intersection)', async () => {
+    const client = psi.client!.create(true)
+    const serverSetup = JSON.stringify({
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      num_hash_functions: 14,
+      bits:
+        '1PnolAVduzqn+tXnoeyZ0ztdnwvAidt0U7gc2rWCtP5i6BDtSUydixXwGHP2mVAZ8PCNqGuTWUQO5VB0OEHTH8Cr1JhLxBDHRM4fAPEUvjjkBPNpsPRRtqzN2fW8y7rcANEwaRyInPsw1t4KYB0Q5vdy38VlPB1/Bz/adfGoGnXQ4fPNI/PN5s+zLcPev9odjAWyUQms8bvPHHsL2uyS7Uu5LRtOfdfYX5zWilxzNXECrabnnUnMqD5NH8c7oreTCWxwRQfYMxhf5K/EZTiTHphCi9O5Ey4ahwvNvykbvXdHQ7CDnLTKmomN6su5BrUt'
+    })
+    const serverResponse = JSON.stringify({
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      encrypted_elements: [
+        'Aw8M6ufg9yUnMdHXQKqZ+2g4Z2uhLzn2J/7dKnp/uDEL',
+        'Av/dJyi/E7gunWLvXX2A1AcnU6dEmB40igRCrqskGTgB',
+        'Av8i8ltywvz4VRFTFju//VoavSL5Fmt95wWd8ChPCrCS',
+        'A1kEdR6hNbGz1sQfHSInQrbZfpDTu6OPiQbnEisAA1wl',
+        'Ap/z+I2cPqyUxM0YYlLOpKtRJYaSMfgNcyRY1Jynun0r',
+        'A6CpgzV4+F6Jjr+yOYbLdpkphMVTjoRuiZOP8uI+7he3',
+        'AnQOkG1+kkYuMvtOcQPqV5YKTYoVe/P+/cUHrfaWDJaD',
+        'An8POEQwL/cEHmQrwAdoBSPYh6a6jPOthH3vBx+2owOF',
+        'A23B3m3hJtAi607Z8FKF/P4fXCTdMCPzOjxgCFlna9K6',
+        'AqV+jTb23uTHPDQmPZsEuxDm5kKYIuI9pkEjvM8fMTG9'
+      ]
+    })
+    const intersection = client.getIntersection(serverSetup, serverResponse)
+    expect(intersection).toStrictEqual([])
```

Comment:
Do we have tests with non-empty intersection? I know this is hard to do without an actual server instance. Maybe this should be done in a new integration.test.ts test suite? For C++ this is currently tested in the server tests, which also isn't ideal.

Reply:
Yes I'll add that. I typically set two folders for unit and integration tests. Since these are relatively small, I'll just put it in the same directory.